### ISSUE TITLE
Support for quoted identifiers

### ISSUE THREAD

**Issue by juhaszp-uhu** [sqlc-dev/sqlc#1587]
### Version

Other

### What happened?

Column names in a table definition may contain characters that are not valid in a Go identifier, yet, this is not recognized and invalid Go code is generated.

In postgresql it is possible to use quoted identifiers in table definitions, and these identifiers may contain any character (apart from \x00) inside the quotes. Relevant documentation: https://www.postgresql.org/docs/current/sql-syntax-lexical.html

It would be great if sqlc supported these. There is already some mapping and reformatting going on between Postgresql column names and Go struct members (e.g. converting to camel case), so I don't see why sanitized, Go-safe names couldn't be generated for these fancy quoted columns.

I get it that it's best to avoid using such names and restrict ourselves to well-behaved identifiers, but sometimes it is not an option. For example, the Asterisk/PJSIP database uses column names beginning with numbers or `@` characters, and in this case we don't have the option to change the schema.

### Relevant log output

```shell
type Foo struct {
  ID int64 
  100beginsWithNumber sql.NullInt32 
  @HasFancyCharacters sql.NullString 
}



# package db
error generating code: source error: 23:3: expected '}', found 100 (and 1 more errors)
```

### Database schema

```sql
create table foo (
	id bigserial primary key,
	"100begins_with_number" integer,
	"@has_fancy_characters" varchar 
);
```

### SQL queries

```sql
-- name: TestQuery :one
select * from foo where id = ?;
```

### Configuration

```yaml
version: "1"
packages:
  - path: "db"
    name: "db"
    engine: "postgresql"
    schema: "schema.sql"
    queries: "queries.sql"
```

### What operating system are you using?

Linux

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by juhaszp-uhu** [comment]
(Version vas 1.13.0, but the form didn't allow that)

**Comment by kyleconroy** [comment]
Do you have a preference for how we support these columns? We already have some "magic" for replace invalid characters in enum names.

I see two options. We can automatically replace the invalid characters or we can require explicit renaming.

**Comment by juhaszp-uhu** [comment]
Thanks for the reply!

I gave this some thought. Just replacing the offending characters with the empty string won't work, because that may result in identical sanitized column names (e.g. "4th_bad_column" vs "5th_bad_column"). Replacing them with something like their Unicode character codes ("4th_bad_column" -> "X0034BadColumn") may work, though.

I also like the option to allow explicit renaming on a column by column basis. That does push the responsibility on the user, so you don't have to invent an arbitrary replacement scheme. 

In any case you have to _detect_ such column names, either to trigger the replacement code path, if you choose that, or to give an appropriate error message. It's doable, though, because the list of valid characters in a Go identifiers is well-defined, so you just have to check for that.

To answer your actual question: perhaps the combination of the two approach would be best, with an automatic replacement scheme, plus a warning message and the capability to rename columns at will. That would be useful even outside the context of this bug.

**Comment by Emyrk** [comment]
I hit this too with this playground: https://play.sqlc.dev/p/44a3d9bcf8a0325e547b9edb7338287e89f22468e2aa9e8f9f86a5caffbaa18b


My use case was using [sqlx](https://github.com/jmoiron/sqlx) which supports nested struct syntax with `.`. I was using sqlc to generate most of my sql, but the model type was not nice to work with, so I made my own model and used sqlx.

**Comment by kyleconroy** [comment]
Fixed as of sqlc 1.21 

```
type Foo struct {
	ID                   int64
	_100beginsWithNumber sql.NullInt32
	HasFancyCharacters   sql.NullString
}
```

https://play.sqlc.dev/p/7cd3522c5607127ed2e5c8a9ccb469912f771f64fe4d2e145b758c636366f6cb

**Comment by Emyrk** [comment]
You still cannot use `.` in the quoted identifier.

> I hit this too with this playground: https://play.sqlc.dev/p/44a3d9bcf8a0325e547b9edb7338287e89f22468e2aa9e8f9f86a5caffbaa18b
